actions/actions.py

```diff
@@ -1,191 +1,199 @@
+# custom Python code.
+#
+# See this guide on how to implement these action:
+# https://rasa.com/docs/rasa/custom-actions
+
+
+# This is a simple example for a custom action which utters "Hello World!"
+
+from typing import Any, Text, Dict, List
+
+from rasa_sdk import Action, Tracker
+from rasa_sdk.executor import CollectingDispatcher
+from rasa_sdk.events import SlotSet
+from rasa_sdk.events import FollowupAction
+import sqlite3
+
+
+class ActionProductSearch(Action):
+    def name(self) -> Text:
+        return "action_product_search"
+
+    def run(
+        self,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> List[Dict[Text, Any]]:
+
+        # connect to DB
+        conn = sqlite3.connect("example.db")
+        c = conn.cursor()
+
+        # get slots
+        shoe = [(tracker.get_slot("color")), (tracker.get_slot("number"))]
+
+        # retrieve row based on search criteria
+        c.execute("SELECT * FROM inventory WHERE color=? AND size=?", shoe)
+        rw = c.fetchone()
+
+        if rw:
+            # provide in stock message
+            dispatcher.utter_message(template="utter_in_stock")
+            conn.close()
+            reset_slots = ["number", "color"]
+            return [SlotSet(slot, None) for slot in reset_slots]
+        else:
+            # provide out of stock
+            dispatcher.utter_message(template="utter_no_stock")
+            conn.close()
+            reset_slots = ["number", "color"]
+            return [SlotSet(slot, None) for slot in reset_slots]
+
+
+class SurveyStart(Action):
+    def name(self) -> Text:
+        return "action_survey_start"
+
+    async def run(
+        self,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> List[Dict[Text, Any]]:
+
+        if tracker.get_slot("survey_complete") == True:
+            return []
+        else:
+            return [FollowupAction("survey_form"), SlotSet("number", None)]
+
+
+class SurveySubmit(Action):
+    def name(self) -> Text:
+        return "action_survey_submit"
+
+    async def run(
+        self,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> List[Dict[Text, Any]]:
+
+        dispatcher.utter_message(template="utter_open_feedback")
+        dispatcher.utter_message(template="utter_survey_end")
+        return [SlotSet("survey_complete", True), SlotSet("number", None)]
+
+
+class OrderStatus(Action):
+    def name(self) -> Text:
+        return "action_order_status"
+
+    def run(
+        self,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> List[Dict[Text, Any]]:
+
+        # connect to DB
+        conn = sqlite3.connect("example.db")
+        c = conn.cursor()
+
+        # get email slot
+        order_email = (tracker.get_slot("email"),)
```

Comment:
```suggestion
        order_email = tracker.get_slot("email")
```

Reply:
I'll try this out again but I had to add the format like this to make sure the tuple formatting was right for the sql query